core/lib/storeCodeFromRoute.ts

```diff
@@ -0,0 +1,49 @@
+import { RawLocation } from 'vue-router'
+import config from 'config'
+import { LocalizedRoute } from './types'
+
+const getNormalizedPath = (matchedRouteOrUrl) => {
+  const matchingPath = typeof matchedRouteOrUrl === 'object' ? matchedRouteOrUrl.path : matchedRouteOrUrl
+
+  return matchingPath.length > 0 && matchingPath[0] !== '/' ? `/${matchingPath}` : matchingPath
```

Comment:
```suggestion
  return matchingPath && matchingPath.length > 0 && matchingPath[0] !== '/' ? `/${matchingPath}` : matchingPath
```

would be nice to have these methods unit tested